interpreter/interpreter_test.go

```diff
@@ -61,11 +61,13 @@ func TestEval(t *testing.T) {
 			wantErr: any,
 		},
 		{
-			name: "string interpolation field has wrong type",
+			name: "string interpolation non-string type",
 			query: `
 				r = makeRecord(o: {a: "foo", b: 42})
 				"r._value = ${r.b}"`,
```

Comment:
Can you update this to have both cases? A case where it errors and a case where it succeeds? For example trying to format `r` in the string would still error, `"r = ${r}"`